x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/EsqlActionBreakerIT.java

```diff
@@ -79,6 +86,17 @@ protected Settings nodeSettings(int nodeOrdinal, Settings otherSettings) {
             .build();
     }
 
+    public static class EsqlTestPluginWithMockBlockFactory extends EsqlPlugin {
+        @Override
+        protected BlockFactoryProvider blockFactoryProvider(
```

Comment:
Can't we do this same thing, but injecting the `MockBlockFactory` directly without the need of the extra "Provider" layer? From what I see, the `BlockFactory` is already being injected, so just injecting the mock should be enough? Maybe with this same `blockFactoryProvider()` override.

It's a "Supplier of Factories", which is a bit :hehe:. If it's absolutely needed, no problem with it. But if we can avoid it... At least I'm not seeing what does it give us exactly 👀

Reply:
Sadly, we can't inject a MockBlockFactory for a BlockFactory as it requires exact class names. I tried an alternative that avoids injecting BlockFactory by passing the ExchangeService, but it's less natural than this approach.